packages/integration-tests/src/EntityManager.clone.test.ts

```diff
@@ -0,0 +1,270 @@
+import { insertAuthor } from "@src/entities/inserts";
+import { EntityConstructor, EntityManager } from "joist-orm";
+import { Author, Book, Comment, Image, ImageType, newAuthor, newBook, newPublisher, Publisher, Tag } from "./entities";
+import { newEntityManager } from "./setupDbTests";
+
+describe("EntityManager", () => {
+  it("can clone entities", async () => {
+    const em = newEntityManager();
+
+    // Given an entity
+    const p1 = newPublisher(em, { name: "p1" });
+    const a1 = new Author(em, { firstName: "a1", publisher: p1 });
+    await em.flush();
+
+    // When we clone that entity
+    const a2 = await em.clone(a1);
+    await em.flush();
+
+    // Then we expect the cloned entity to have the same properties as the original
+    expect(a2.firstName).toEqual(a1.firstName);
+    expect(a2.publisher.idOrFail).toEqual(p1.id);
+    expect(a2.id).not.toEqual(a1.id);
+    expect(await numberOf(em, Author, Publisher)).toEqual([2, 1]);
+    expect(p1.authors.get).toEqual([a1, a2]);
+  });
+
+  it("can clone entities and referenced entities", async () => {
+    const em = newEntityManager();
+
+    // Given an entity with a reference to another entity
+    const a1 = newAuthor(em, { firstName: "a1" });
+    const b1 = newBook(em, { title: "b1", author: a1 });
+    // And the author itself points to the book we'll clone
+    a1.currentDraftBook.set(b1);
+    await em.flush();
+
+    // When we clone that entity and its reference
+    const a2 = await em.clone(a1, { hint: "books" });
```

Comment:
FYI - This is a diff from the moved test. `hint` is now in the `opts` instead of the second argument. Seemed like the best path forward with `hint` already being an optional argument. Open to other signature suggestions.

Reply:
👍